tests/qemu-tests/run_qemu_system.sh

```diff
@@ -37,28 +36,21 @@ while [[ $# -gt 0 ]]; do
     shift
 done
 
-CWD=$(dirname -- "$0")
-IMAGE_DIR="${CWD}/images"
-
-if [ -z "$ARCH" ]; then
+if [ -z "${KERNEL_NAME}" ]; then
     help_and_exit
 fi
 
-if [[ ! " ${VALID_ARCHS[*]} " =~ " ${ARCH} " ]]; then
-    echo "Invalid arch '${ARCH}'"
+if [[ ! " ${KERNEL_LIST[*]} " =~ " ${KERNEL_NAME} " ]]; then
+    echo "Invalid kernel '${KERNEL_NAME}'"
     help_and_exit
 fi
 
-if [[ ! " ${VALID_TYPES[*]} " =~ " ${KERNEL_TYPE} " ]]; then
-    echo "Invalid kernel type '${KERNEL_TYPE}'"
-    help_and_exit
-fi
+# extract architecture as last dash-separated group of the kernel's name
+ARCH="${KERNEL_NAME##*-}"
```

Comment:
Can you add a comment here indicating what we expect kernel names to look like? i.e `Image-${KERNEL_TYPE}-${VERSION}-${ARCH}` or whatever it's supposed to be?